src/utils/utils.mjs

```diff
@@ -0,0 +1,54 @@
+export function filterSeancesByFeatures(seances, emptyAmount, hasEmptyVip, hasEmptyDouble) {
+  let filterList = [];
+  emptyAmount && filterList.push(emptySeatsFilter);
+  hasEmptyVip && filterList.push(emptyVipFilter);
+  hasEmptyDouble && filterList.push(emptyDoubleFilter);
+
+  return filterList.reduce((acc, filter) => {
+    acc = filter(acc, emptyAmount);
+    return acc;
+  }, seances);
+}
+
+function emptyVipFilter(seances, emptyAmount) {
+  return seances.filter(seance => {
+    let soldSeats = [];
+    const emptySeats =
+      seance.hallId.rows.reduce((sum, row) => {
+        if (row.rowType === 'vip') {
+          sum += row.rowLength;
+          const currentRow = seance.soldSeats.filter(({ rowNumber }) => rowNumber === row.rowNumber);
+          soldSeats = [...soldSeats, ...currentRow];
+        }
+        return sum;
+      }, 0) - soldSeats.length;
+    return emptySeats > emptyAmount;
+  });
+}
+
+function emptyDoubleFilter(seances, emptyAmount) {
+  return seances.filter(seance => {
+    let soldSeats = [];
+    const emptySeats =
+      seance.hallId.rows.reduce((sum, row) => {
+        if (row.rowType === 'double') {
+          sum += row.rowLength;
+          const currentRow = seance.soldSeats.filter(({ rowNumber }) => rowNumber === row.rowNumber);
+          soldSeats = [...soldSeats, ...currentRow];
```

Comment:
reduce should not influence anything outside callback

Reply:
fixed